shared/utility/io/uart.cpp

```diff
@@ -37,7 +37,7 @@ namespace utility::io {
         }
     }
 
-    void uart::close() {
+    void uart::close() const {
```

Comment:
This function should set `fd` to `-1` after we call `::close(fd)` so that we know the file is not open any more